opentelemetry-exporter-geneva/geneva-uploader/src/ingestion_service/mod.rs

```diff
@@ -0,0 +1,242 @@
+        let config_major_version = env::var("GENEVA_CONFIG_MAJOR_VERSION")
+            .expect("GENEVA_CONFIG_MAJOR_VERSION is required")
+            .parse::<u32>()
+            .expect("GENEVA_CONFIG_MAJOR_VERSION must be a u32");
+        let source_identity = env::var("GENEVA_SOURCE_IDENTITY")
+            .unwrap_or_else(|_| "Tenant=Default/Role=Uploader/RoleInstance=devhost".to_string());
+        let schema_ids =
+            "c1ce0ecea020359624c493bbe97f9e80;0da22cabbee419e000541a5eda732eb3".to_string(); //TODO make this configurable
+
+        // === 3. Define uploader config ===
+        let uploader_config = GenevaUploaderConfig {
+            namespace: namespace.clone(),
+            source_identity,
+            environment: environment.clone(),
+            schema_ids,
+        };
+
+        let config = GenevaConfigClientConfig {
+            endpoint,
+            environment,
+            account,
+            namespace,
+            region,
+            config_major_version,
+            auth_method: AuthMethod::Certificate {
+                path: cert_path,
+                password: cert_password,
+            },
+        };
+
+        // === 4. Build client and uploader ===
+        let config_client =
+            GenevaConfigClient::new(config).expect("Failed to create config client");
+        let uploader =
+            match GenevaUploader::from_config_client(&config_client, uploader_config).await {
+                Ok(u) => u,
+                Err(e) => {
+                    eprintln!("Error creating uploader: {:?}", e);
+                    panic!("Failed to create uploader");
+                }
+            };
+
+        // === 5. Upload ===
+        let event_name = "Log";
+        let event_version = "Ver2v0";
+        println!("🚀 Uploading to: {}", uploader.auth_info.endpoint);
+
+        let start = Instant::now();
+        let response = uploader
+            .upload(data, event_name, event_version)
+            .await
+            .expect("Upload failed");
+
+        println!(
+            "✅ Upload complete in {:.2?}. Ticket: {}",
+            start.elapsed(),
+            response.ticket
+        );
+    }
+
+    /// To run this test with parallel uploads:
+    ///
+    /// ```bash
+    /// export GENEVA_ENDPOINT="https://<gcs-endpoint>"
+    /// export GENEVA_ENVIRONMENT="Test"
+    /// export GENEVA_ACCOUNT="YourAccount"
+    /// export GENEVA_NAMESPACE="YourNamespace"
+    /// export GENEVA_REGION="YourRegion"
+    /// export GENEVA_CONFIG_MAJOR_VERSION="2"
+    /// export GENEVA_CERT_PATH="/path/to/client.p12"
+    /// export GENEVA_CERT_PASSWORD="your-password"
+    /// export GENEVA_SOURCE_IDENTITY="Tenant=YourTenant/Role=Role/RoleInstance=Instance"
+    /// export GENEVA_BLOB_PATH="/path/to/blob.bin"
+    /// export GENEVA_PARALLEL_UPLOADS="10"
+    ///
+    /// cargo test test_parallel_uploads -- --ignored
+    /// ```
+
+    #[tokio::test]
```

Comment:
The default test runtime is single threaded. Not required for this PR, but I'm curious if you would see an improvement with multi-threaded runtime and at what point (for what value of `parallel_uploads`).

Reply:
The plan is to have ideal solution to have the SDK scale with parallel_uploads equal to the CPU core. However the current benchmark test is not the right approach to measure, as it adds the network latency. Will be later adding better test mocking the server.